pkg/k8s/function_status.go

```diff
@@ -48,8 +51,32 @@ func AsFunctionStatus(item appsv1.Deployment) *types.FunctionStatus {
 	for _, v := range functionContainer.Env {
 		if EnvProcessName == v.Name {
 			function.EnvProcess = v.Value
+			continue
 		}
+		function.EnvVars[v.Name] = v.Value
 	}
 
 	return &function
 }
+
+func nodeSelectorToConstraint(deploy appsv1.Deployment) []string {
+	nodeSelector := deploy.Spec.Template.Spec.NodeSelector
+	constraints := []string{}
+	for k, v := range nodeSelector {
+		constraints = append(constraints, k+"="+v)
+	}
+	return constraints
+}
+
+func hasReadOnlyRootFilesystem(function appsv1.Deployment) bool {
+	securityContext := function.Spec.Template.Spec.Containers[0].SecurityContext
```

Comment:
Could it be indexed upon some kind of property or name in a range loop?

Reply:
Yes, it can be `nil`, but this is checked immediately on the next line.

Reply:
regarding the position of the container:  

Both the `controller` and the `handler` put the labels on the Pod, but not the container. We can use the container `Name` though, it  will equal the `deployment.Name`.

As a side note, it would be really nice if we could refactor the an unify the generation of the Deployment, the function Factory doesn't actually generate it, both the controller and the handler have different code for the construction of the Deployment

Reply:
@alexellis i have updated the code and replaced all instances of `Containers[0]` with a proper `FunctionContainer()` implementation that finds the correct index and container spec.